src/coreclr/src/tools/crossgen2/ILCompiler.Reflection.ReadyToRun/R2RSignature.cs

```diff
@@ -1054,10 +1054,6 @@ private void ParseType(StringBuilder builder)
                     builder.Append("var_zapsig");
                     break;
 
-                case CorElementType.ELEMENT_TYPE_NATIVE_ARRAY_TEMPLATE_ZAPSIG:
```

Comment:
This code path is used to parse signatures in ready-to-run images.

As far as I understand, ready-to-run images are supposed to be forward compatible. Old crossgen compiled ready-to-run images should run on a newer version of the runtime. Changing the signature format seems to break things.

@VSadov @jkotas 

At the very least, R2RDump should still be able to dump the content of the older version of ready to run images, thoughts?